File: Android_Studio/Java/Cara_ou_Coroa/app/src/main/java/com/example/caraoucoroa/MainActivity.java
package com.example.caraoucoroa;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import java.util.Random;

public class MainActivity extends AppCompatActivity
{
    private Button botaoJogar;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        botaoJogar = findViewById(R.id.botaoJogar);

        botaoJogar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), SegundaActivity.class);

                // Passar dados pra próxima tela
                int numero = new Random().nextInt(2);
                intent.putExtra("numero", numero);

                startActivity(intent);
            }
        });
    }
}

File: Android_Studio/Kotlin/Cara_ou_Coroa/app/src/main/java/com/example/caraoucoroa/MainActivity.kt
package com.example.caraoucoroa

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.random.Random

class MainActivity : AppCompatActivity()
{

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        image_jogar.setOnClickListener {
            val intent = Intent(applicationContext, SegundaActivity:: class.java)


            // Gerar dados randômicos
            val numero = Random.nextInt(2)

            // Enviar número para a próxima activity
            intent.putExtra("numero", numero)

            startActivity(intent)
        }
    }
}
